### PR TITLE
make constants explicit in synchronization database

### PR DESCRIPTION
**Pull request by johnwlambert**
Make the frame rate constants more readable/ clear.

### CODE REVIEW

File: argoverse/data_loading/synchronization_database.py [hunk 1]
```diff
@@ -14,6 +14,17 @@
 
 logger = logging.getLogger(__name__)
 
+RING_CAMERA_FPS = 30
+STEREO_CAMERA_FPS = 5
+LIDAR_FRAME_RATE_HZ = 10
+
+# in milliseconds
+RING_CAMERA_SHUTTER_INTERVAL_MS = (1 / RING_CAMERA_FPS) * 1000 # evaluates to 33.3 ms
```
Comment (benjaminrwilson) [inline argoverse/data_loading/synchronization_database.py]: I think that 1000 might be more readable as `1e3`.

Comment (benjaminrwilson) [reply argoverse/data_loading/synchronization_database.py]: (in terms of quickly determining the units conversion)

File: argoverse/data_loading/synchronization_database.py [hunk 2]
```diff
@@ -14,6 +14,17 @@
 
 logger = logging.getLogger(__name__)
 
+RING_CAMERA_FPS = 30
```
Comment (benjaminrwilson) [inline argoverse/data_loading/synchronization_database.py:17]: Can we add typing here?

File: argoverse/data_loading/synchronization_database.py [hunk 3]
```diff
@@ -59,21 +70,20 @@ def find_closest_integer_in_ref_arr(query_int: int, ref_arr: np.ndarray) -> Tupl
...
-    # LiDAR is 10 Hz
-    # Max we are halfway between 33.3 milliseconds on either side
+    # Max difference between camera and LiDAR observation would be if the LiDAR timestamp is halfway between
+    # two camera observations (i.e. RING_CAMERA_SHUTTER_INTERVAL_MS / 2 milliseconds on either side)
     # then convert milliseconds to nanoseconds
-    MAX_LIDAR_RING_CAM_TIMESTAMP_DIFF = 33.3 * (1.0 / 2) * (1.0 / 1000) * 1e9
-    # Stereo Camera is 5 Hz (once per 200 milliseconds)
-    # LiDAR is 10 Hz
-    # Since Stereo is more sparse, we look for 200 millisecond on either side
+    MAX_LIDAR_RING_CAM_TIMESTAMP_DIFF = RING_CAMERA_SHUTTER_INTERVAL_MS * (1.0 / 2) * (1.0 / 1000) * 1e9
```
Comment (benjaminrwilson) [inline argoverse/data_loading/synchronization_database.py:77]: I think we should consider making this more readable. One idea:

`MAX_LIDAR_RING_CAM_TIMESTAMP_DIFF = 0.5 * (1e9 / 1e3) * RING_CAMERA_SHUTTER_INTERVAL_MS`

I think ideally we would have a function to change units:

`to_metric_time(ts: int, src: TimeUnit, dst: TimeUnit)`

or something similar.

**Review by benjaminrwilson — CHANGES_REQUESTED** [review]

File: argoverse/data_loading/synchronization_database.py (hunk 2 above)
Comment (johnwlambert) [reply argoverse/data_loading/synchronization_database.py:17]: typing_extensions.Final?

File: argoverse/data_loading/synchronization_database.py (hunk 3 above)
Comment (johnwlambert) [reply argoverse/data_loading/synchronization_database.py:77]:
```python
MAX_LIDAR_RING_CAM_TIMESTAMP_DIFF = to_metric_time(RING_CAMERA_SHUTTER_INTERVAL_MS / 2), src=Millisecond, dst=Nanosecond)
```

**Pull request merged**